### PR TITLE
options/internal: Improve float parsing

### PR DESCRIPTION
**Pull request by Qwinci**
Support INF/INFINITY/nan/NAN, hex floats and exponents.

### CODE REVIEW

**Review by Dennisbonke — APPROVED** [review]
LGTM

File: options/internal/include/mlibc/strtofp.hpp [hunk 1]
```diff
@@ -3,43 +3,152 @@
...
+			return __builtin_infl();
+	} else if (strcmp(str, "INFINITY") == 0 || strcmp(str, "infinity") == 0) {
+	if (endptr)
+		*endptr = (char *)str + 8;
+	if constexpr (std::is_same_v<T, float>)
+		return __builtin_inff();
+	else if constexpr (std::is_same_v<T, double>)
+		return __builtin_inf();
+	else
+		return __builtin_infl();
```
Comment (qookei) [inline options/internal/include/mlibc/strtofp.hpp:29]: Not indented enough

**Review by qookei — COMMENTED** [review]
I think the logic should be fine.

File: options/internal/include/mlibc/strtofp.hpp (hunk 1 above)
Comment (Dennisbonke) [reply options/internal/include/mlibc/strtofp.hpp:29]: Damn, missed that. My bad.

**Pull request merged**